main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -41,6 +41,8 @@ sealed trait BackendObjType {
     case BackendObjType.Lazy(tpe) => JvmName(RootPackage, mkClassName("Lazy", tpe))
     case BackendObjType.Ref(tpe) => JvmName(RootPackage, mkClassName("Ref", tpe))
     case BackendObjType.Tuple(elms) => JvmName(RootPackage, mkClassName("Tuple", elms))
+    case BackendObjType.Tagged => JvmName(RootPackage, mkClassName("Tagged"))
+    case BackendObjType.Tag(tpe) => JvmName(RootPackage, mkClassName("Tag", tpe))
```

Comment:
What is the difference between Tagged and Tag?

Reply:
Tagged is something that has a `tag: String`. `Tag[t]` is indexed, and has a value of type `t`

Reply:
I don't get it.

Reply:
```
abstract class Tagged { public String name; }
class Tag$Int32 extends Tagged { public int value }
class Tag$Float32 extends Tagged { public float value }
```

So when you want to compile `e Is `enum.case`, we don't know yet the exact type (it might return false), so we need a common interface on which to ask what its case name is. It becomes `e.name == enum.case`

Reply:
Right ok